src/configobj/__init__.py

```diff
@@ -1032,18 +1033,21 @@ class ConfigObj(Section):
     # TODO: also support inline comments (needs dynamic compiling of the regex below)
     COMMENT_MARKERS = ['#']
 
-    _keyword = re.compile(r'''^ # line start
-        (\s*)                   # indentation
-        (                       # keyword
-            (?:".*?")|          # double quotes
-            (?:'.*?')|          # single quotes
-            (?:[^'"=].*?)       # no quotes
-        )
-        \s*=\s*                 # divider
-        (.*)                    # value (including list values and comments)
-        $   # line end
-        ''',
-        re.VERBOSE)
+
+    @property
+    def _keyword(self):
+        return re.compile(r'''^ # line start
+            (\s*)                   # indentation
+            (                       # keyword
+                (?:".*?")|          # double quotes
+                (?:'.*?')|          # single quotes
+                (?:[^'"{0}].*?)     # no quotes
+            )
+            \s*[{0}]\s*             # dividers
+            (.*)                    # value (including list values and comments)
+            $   # line end
+            '''.format(''.join(self._dividers)),
+            re.VERBOSE)
```

Comment:
This breaks a unit test: `TestMerging.test_merge_coupling[data1]` as you cannot deep copy the object then.